### ISSUE TITLE
fix: @Validateif does not work and does not report an error

### ISSUE THREAD

**Issue by MobroChu** [typestack/class-validator#1271]
### Description
The custom validation rule '@ validateif' does not take effect. When this function returns false, I want to give Fe feedback that the parameters are incorrect.
The 'class validator' version is 0.13.1, and the reduced version is 0.12.1, which is unsuccessful.

**Minimal code-snippet showcasing the problem**
 

Dto files are as follows
```ts
import { Iobject } from '@mobro/libs/lib/type';
import { IsNotEmpty, IsNumber, IsOptional, IsString, ValidateIf } from 'class-validator';

export class CreateActivityDto {
    @ValidateIf(
        (obj: Iobject, val: number) => {
            console.log(val, 'obj----val', obj);
            return false;
        },
        {
            message: '随便提示点什么'
        }
    )
    @IsNotEmpty({ message: '活动开始时间必填' })
    @IsNumber({}, { message: '活动开始时间应为一个时间戳' })
    startTime: number;
}
```

**Comment by julien-sarazin** [comment]
Same issue.
